### PR TITLE
USHIFT-355: MicroShift: skip sig-network tests

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

File: test/extended/util/framework.go [hunk 1]
```diff
@@ -2071,3 +2083,37 @@ func DoesApiResourceExist(oc *CLI, apiResourceName string) (bool, error) {
 
 	return false, nil
 }
+
+var (
+	microShiftVersion string
+	microShiftMutex   sync.Mutex
```
Comment (jaypoulz) [inline test/extended/util/framework.go:2089]: What is this protecting against?

**Comment by pacevedom** [comment]
/hold
Need to get rid of microshift checks.

File: test/extended/idling/idling.go [hunk 2]
```diff
@@ -239,7 +239,7 @@ var _ = g.Describe("[sig-network-edge][Feature:Idling]", func() {
 		})
 	})
 
-	g.Describe("Unidling", func() {
+	g.Describe("Unidling [apigroup:apps.openshift.io]", func() {
```
Comment (ingvagabund) [inline test/extended/idling/idling.go:242]: `route.openshift.io` as well (both apps and route are used in `testdata/idling-echo-server.yaml` (set through `echoServerFixture`)

Comment (ingvagabund) [reply test/extended/idling/idling.go:242]: Also `with a single service and DeploymentConfig` (line 218) is using `route.openshift.io` (through `testdata/idling-echo-server-rc.yaml`)

File: test/extended/router/stress.go [hunk 3]
```diff
@@ -71,7 +71,9 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:route.openshift.io]",
 				Kind: "ClusterRole",
 				Name: "system:router",
 			},
-		}, metav1.CreateOptions{})
+		}
+
+		_, err = oc.AdminKubeClient().RbacV1().RoleBindings(ns).Create(context.Background(), roleBinding, metav1.CreateOptions{})
```
Comment (ingvagabund) [inline test/extended/router/stress.go:76]: Why this change when `roleBinding` is used only once?

File: test/extended/util/framework.go [hunk 4]
```diff
@@ -1949,6 +1949,17 @@ func GetRouterPodTemplate(oc *CLI) (*corev1.PodTemplateSpec, string, error) {
 }
 
 func FindRouterImage(oc *CLI) (string, error) {
+	microShift, err := IsMicroShiftDeployment(oc)
```
Comment (ingvagabund) [inline test/extended/util/framework.go:1952]: Leftover? `IsMicroShiftDeployment` is never defined.

File: test/extended/idling/idling.go (hunk 2 above)
Comment (pacevedom) [reply test/extended/idling/idling.go:242]: Done

File: test/extended/util/framework.go (hunk 4 above)
Comment (pacevedom) [reply test/extended/util/framework.go:1952]: Done

File: test/extended/router/stress.go (hunk 3 above)
Comment (pacevedom) [reply test/extended/router/stress.go:76]: Leftover from previous change.

File: test/extended/util/framework.go [hunk 5]
```diff
@@ -1949,6 +1949,17 @@ func GetRouterPodTemplate(oc *CLI) (*corev1.PodTemplateSpec, string, error) {
 }
 
 func FindRouterImage(oc *CLI) (string, error) {
+	exists, err := DoesApiResourceExist(oc, "clusteroperators")
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		deployment, err := oc.AdminKubeClient().AppsV1().Deployments("openshift-ingress").Get(context.Background(), "router-default", metav1.GetOptions{})
```
Comment (ingvagabund) [inline test/extended/util/framework.go:1957]: In 4.12 taking `deployment.Spec.Template.Spec.Containers[0].Image` from the deployment gets me `quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a955bf1686fcb4f11cd23880be6025c2dc75252d318931211554be86f4f5d0f1`. Taking `.Status.Versions[*].Version` from clusteroperator/ingress I get `quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a955bf1686fcb4f11cd23880be6025c2dc75252d318931211554be86f4f5d0f1`. @pacevedom is it always guaranteed the images will be identical?

Comment (ingvagabund) [reply test/extended/util/framework.go:1957]: @openshift/networking can you confirm accessing `deployment.Spec.Template.Spec.Containers[0].Image` of `router-default` deployment gets the same image as from the `ingress` clusteroperator object?

Comment (Miciah) [reply test/extended/util/framework.go:1957]: Yes, `deployment.Spec.Template.Spec.Containers[0].Image` of the `openshift-ingress/router-default` deployment is the same image as the "ingress-controller" image in the `ingress` clusteroperator, on both MicroShift and standard OpenShift.  This approach looks reasonable.

Comment (pacevedom) [reply test/extended/util/framework.go:1957]: This is output from a 4.12 cluster:
```
$ oc get deployment -n openshift-ingress router-default -o jsonpath='{.spec.template.spec.containers[*].image}'
quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:72c751aa148bf498839e6f37b304e3265f85af1e00578e637332a13ed9545ece

$ oc get clusteroperator ingress -o jsonpath='{.status.versions[?(@.name=="ingress-controller")].version}'
quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:72c751aa148bf498839e6f37b304e3265f85af1e00578e637332a13ed9545ece
```

`router-default` is MicroShift equivalent for `ingress-controller` pod in OCP.

**Comment by ingvagabund** [comment]
/lgtm
based on https://github.com/openshift/origin/pull/27428#discussion_r986852674

**Comment by ingvagabund** [comment]
/hold
until https://github.com/openshift/origin/pull/27416 gets merged

File: test/extended/idling/idling.go [hunk 6]
```diff
@@ -216,7 +216,7 @@ var _ = g.Describe("[sig-network-edge][Feature:Idling]", func() {
 	})
 
 	g.Describe("Idling", func() {
-		g.Context("with a single service and DeploymentConfig", func() {
+		g.Context("with a single service and DeploymentConfig [apigroup:route.openshift.io]", func() {
```
Comment (deads2k) [inline test/extended/idling/idling.go:219]: also apps.openshift.io

Comment (pmtk) [reply test/extended/idling/idling.go:219]: apps.openshift.io is already present in g.It:
```
g.It("should idle the service and DeploymentConfig properly [apigroup:apps.openshift.io]", func() {
```
do we need to add it here as well?

Comment (ingvagabund) [reply test/extended/idling/idling.go:219]: We don't need to given the fact.

File: tools/junitreport/pkg/parser/gotest/data_parser_test.go [hunk 7]
```diff
@@ -161,33 +161,33 @@ func TestExtractSuiteName(t *testing.T) {
 		fail         bool
 	}{
 		{
-			name:         "basic",
-			testLine:     "ok  	package/name 0.160s",
+			name: "basic",
```
Comment (ingvagabund) [inline tools/junitreport/pkg/parser/gotest/data_parser_test.go:164]: Looks like this is formatted with go1.18. @pacevedom the codebase has been recently formatted with go1.19.

**Comment by ingvagabund** [comment]
`[sig-network-edge][Feature:Idling] Unidling [apigroup:apps.openshift.io][apigroup:route.openshift.io] should handle many TCP connections by possibly dropping those over a certain bound [Serial] [Suite:openshift/conformance/serial] ` is constantly failing. Because it's originally skipped by the name changed: https://github.com/openshift/origin/blob/master/test/extended/util/annotate/rules.go#L68

**Comment by ingvagabund** [comment]
Merged through https://github.com/openshift/origin/pull/27498
